### PR TITLE
Add .gitlab-ci.yml and pre-cache script

### PR DESCRIPTION
**Pull request by alvicsam**
I mirrored the repo to Gitlab and now setting up CI/CD.

Closes https://github.com/paritytech/ci_cd/issues/225

### CODE REVIEW

**Comment by gilescope** [comment]
Looking good. Can we do a cargo unleash to deploy to crates.io when a new version is tagged?

**Comment by alvicsam** [comment]
Yes, sure. I'm in progress :)

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -2,13 +2,26 @@ stages:
...
+  - publish
 
 variables:                         &default-vars
   GIT_STRATEGY:                    fetch
   GIT_DEPTH:                       100
   CARGO_INCREMENTAL:               0
   CARGO_TARGET_DIR:                "/ci-cache/${CI_PROJECT_NAME}/targets/${CI_COMMIT_REF_NAME}/${CI_JOB_NAME}"
   CI_IMAGE:                        "paritytech/ci-linux:production"
+  CARGO_UNLEASH_INSTALL_PARAMS:    "--version 1.0.0-alpha.12"
+  CARGO_UNLEASH_PKG_DEF:           "--skip node node-* pallet-template pallet-example pallet-example-* subkey chain-spec-builder"
```
Comment (gilescope) [inline .gitlab-ci.yml:14]: this can probably be an empty list?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  <<:                              *build-refs
+  <<:                              *docker-env
+  script:
+    - cargo build --no-default-features --target wasm32-unknown-unknown --verbose
+
+#### stage:                       publish
+
+unleash-to-crates-io:
+  stage:                           publish
+  <<:                              *docker-env
```
[the remark this answers is not in the document]
Comment (alvicsam) [reply .gitlab-ci.yml:130]: Oh, okay, I'll change to k8s runner then

**Review by rcny — APPROVED** [review]

**Comment by rcny** [comment]
Ok, thanks, let's keep `linux-docker`.

**Pull request merged**